common/sgx/qeidentity.c

```diff
@@ -57,18 +57,29 @@ oe_result_t oe_enforce_qe_identity(sgx_report_body_t* qe_report_body)
         // enclave's mrsigner.
         if (!oe_constant_time_mem_equal(
                 qe_report_body->mrsigner, g_qe_mrsigner, sizeof(g_qe_mrsigner)))
-            OE_RAISE_MSG(OE_VERIFY_FAILED, "mrsigner mismatch", NULL);
+            OE_RAISE_MSG(
+                OE_QUOTE_ENCLAVE_IDENTIFY_VERIFICATION_FAILED,
+                "mrsigner mismatch",
+                NULL);
 
         if (qe_report_body->isvprodid != g_qe_isvprodid)
-            OE_RAISE_MSG(OE_VERIFY_FAILED, "isvprodid mismatch", NULL);
+            OE_RAISE_MSG(
+                OE_QUOTE_ENCLAVE_IDENTIFY_VERIFICATION_FAILED,
+                "isvprodid mismatch",
+                NULL);
 
         if (qe_report_body->isvsvn < g_qeisvsvn)
-            OE_RAISE_MSG(OE_VERIFY_FAILED, "isvsvn is out-of-date", NULL);
+            OE_RAISE_MSG(
+                OE_QUOTE_ENCLAVE_IDENTIFY_VERIFICATION_FAILED,
+                "isvsvn is out-of-date",
```

Comment:
OE_QUOTE_ENCLAVE_IDENTIFY_VERIFICATION_FAILED error code is shared by above three error cases. 
It would be easier to root cause if distinct error codes are used.

Reply:
It would also probably help application operators/end-users to include the expected and found isvsvn and isvprodid values in the error messages.

Reply:
Added 3 additional error codes.

Reply:
@jazzybluesea thank you!